### DISCUSSION TITLE
MJWSD05MMC binary

### DISCUSSION THREAD

**Discussion by w00dst0ck** [pvvx/ZigbeeTLc#41] [General]
Happy new year!

Do you have any idea when a release for the MJWSD05MMC could be expected?

Best regards and thank you for your work.

**Comment by pvvx** [comment]
For Zigbee MJWSD05MMC is not interesting. ZHA/Z2M cannot work with Zigbee 3.0. I will not rewrite ZHA/Z2M to support all the features of MJWSD05MMC.
When Z2M/ZHA become user-friendly and automatically support all functions, then it will be possible to write Zigbee firmware for MJWSD05MMC

Works well on BLE Long Range:

![image](https://github.com/pvvx/ZigbeeTLc/assets/12629515/a00d3f89-85ef-443a-8245-a206ce4c0b25)
The battery drains very slowly.

Neighboring house, 200 meters.

**Comment by TheMonkeyBrain** [comment]
I can see that [SMLIGHT SLZB-06M](https://smlight.tech/product/slzb-06m/) and other Zigbee 3.0 devices are supported by both ZHA/Z2M.
What exactly is missing/needed to write a zigbee firmware for MJWSD05MMC?

I also notiched that the [README.md](https://github.com/pvvx/ZigbeeTLc/blob/master/README.md) specifies MJWSD05MMC as 'in developing'.

Thanks for your great work.

**Comment by pvvx** [comment]
> What exactly is missing/needed to write a zigbee firmware for MJWSD05MMC?

1. There is no desire to spoil a good thermometer with a hopeless and power-hungry version of the ZigBee protocol.
2. ZHA and Z2M have not become more user-friendly over the past year and do not have all the services.
3. With the release of Bluetooth 5.4, the Zigbee protocol becomes unnecessary and obsolete.
4. Due to Linus Torvalds' political views and sanctions against Russia, it is no longer possible to use the Linux kernel, which has not been developed for 10 years in the field of IoT. I am moving to more modern operating systems that are not available in other countries. But I am not stopping support for older versions of BLE/ZigBee sensors in legacy systems and will continue to do so as they evolve.

---
> I can see that [SMLIGHT SLZB-06M](https://smlight.tech/product/slzb-06m/) and other Zigbee 3.0 devices are supported by both ZHA/Z2M.

To communicate with devices at a distance of 200 meters in my area, BLE in the "LE Long Range" mode is enough for me. Repeaters and extenders are not required to cover a larger range.
Zigbee and WiFi are not suitable - they cannot work reliably over such distances at reception levels below -95 dBm, as "LE Long Range" allows ( for the cheapest USB-BT adapter RLT8761BU without an external antenna).
![image](https://github.com/user-attachments/assets/697f522a-3428-4c70-aa31-2bc40114863c)
![image](https://github.com/user-attachments/assets/4845a365-f9b0-4eb7-b722-38acd6ccfbd9)

And my automation system for maintaining temperature is not satisfied with the slow readings of Zigbee sensors:
![image](https://github.com/user-attachments/assets/d6f83878-2057-4500-bf07-61a97004f2c3)

The same applies to motion sensors and light switches. The response of ZigBee devices is slow and exceeds comfort standards. The response to pressing a button or other human action should not exceed 200 ms. In my BLE devices, event processing takes only 3..7 ms.

**Reply by TheMonkeyBrain** [reply]
Thank you for the detailed explanation.

**Reply by w00dst0ck** [reply]
@TheMonkeyBrain 
I now use an ESP32 as a Bluetooth proxy in Homeassistant for this purpose.
https://www.home-assistant.io/integrations/bluetooth/#remote-adapters-bluetooth-proxies

**Reply by Benik3** [reply]
What exact BLE LE LR temperature sensor do you use?
I just installed 3 LYWSD03MMC with Zigbee firmware in my parents house and I'm pretty disappointed by the signal (even that with WiFi coverage I have no problem).
Thanks!

**Reply by pvvx** [reply]
> What exact BLE LE LR temperature sensor do you use?

Any of https://github.com/pvvx/ATC_MiThermometer

For example, LYWSD03MMC B1.4:
![image](https://github.com/user-attachments/assets/745679d0-7ad9-44f4-88fe-d029ba3fccc3)
This reception level is not achievable with Zigbee (-100 dBm max) or WiFi (-95 dBm max).
Since June 16, 2016, Bluetooth 5.0 was released and the range of BLE was increased fourfold...
The TLSR825x and PHY62x2 chips, like many others, including BT adapters, support LE LR since 2016. But not Linux.

And starting with Bluetooth v5.4, Zigbee can be considered completely obsolete. Only this will not affect Linux users. Only Bluetooth v4.2 will be supported there.
Android smartphones have long supported LE LR at the native API level. The new functionality in BT5.4 is an additional software implementation and is already available in some Android smartphones...

**Reply by Benik3** [reply]
Thanks. Yeah, I found that HA use bluez BT stack, that doesn't support BT 5.0 yet. Do you have any recommendation for BT sensor implementation to HA? I have it on RPi 4B... Maybe some converter? I found that e.g. ESP32-H3 support bluetooth 5.4.

BTW how is the Zigbee link quality calculated and what is considered as bad? I know that it's number 0-255, but meaning of the number depends on the specific integration. On one sensor I have quality 75-80.

EDIT: It looks like ESPHome BT proxy can be option. ESP32-C6 is also possible to get to work.
I just would like to have some more "hard wired" connection of the proxy then WiFi...

Can I flash your BT firmware using zigbee OTA update or I have to use UART method?

**Reply by pvvx** [reply]
> Yeah, I found that HA use bluez BT stack, that doesn't support BT 5.0 yet. Do you have any recommendation for BT sensor implementation to HA?

https://github.com/pvvx/ATC_MiThermometer/issues/297

> I have it on RPi 4B...
> ESP32-H3 support bluetooth 5.4.

The description that the chip supports BT5.0+ does not mean that it has software support.
